overlay point support and other improvements and fixes

AndrewArace opened the pull request:
Added floorplan.maxZoom() getter/setter, for modifying the depth you
could zoom in (i was using large detailed SVG backdrops for the imageLayer)

Fixed project name in build.xml (I don't know what "position-analytics" was, but I assume it was a copy-paste typo from another project)

Recompiled min.js

(sorry for the tabs/spaces diffs)

aarace committed:
added maxZoom modifier

Added floorplan.maxZoom() getter/setter, for modifying the depth you
could zoom in.
Fixed project name in build.xml
Recompiled min.js

ghost commented:
If you accept it, I'll handled adding to the API wiki page.

aarace committed:
Point support

floorplan - added layersWidth() assignment to make map tools window
bigger
overlay - added support for standalone points
overlay - fixed bug in select callback
overlay - added dragging logic for click versus drag
rebuilt min.js

ghost commented:
floorplan - added layersWidth() assignment to make map tools window
bigger
overlay - added support for standalone points
overlay - fixed bug in select callback
overlay - added dragging logic for click versus drag
rebuilt min.js

aarace committed:
multi layer support

Changed date-based id scheme to contain internal identifier to allow for
multiple instances of same layer types (multiple image layers, for
example)

Added visible() function to layer definition to change initial state of
layer visibility on stage load

dciarletta commented:
Thank you for the contributions. I like all the changes you made. Although in the addition of the point support it appears you may have changed the display of the polygons when not in editMode. Originally if editMode is off the polygons would render without the points but I think that is not the case now. I apologize if I'm incorrect, but I didn't have time to test that theory out so I'm only going on code inspection at this point. If you don't get to it before I do I'll confirm if that is the case and patch it when I accept your pull request.

ghost commented:
Hi,
Thanks for reviewing this - I did this in a relatively fast timeline for a small demo project, and figured I'd pass along some of my enhancements, but I admit that my changes may have impacted other functionality without meaning to.
I won't be working on this demo project any more (actually moving to a new company), so it's up to you if you believe the changes are worth me looking into and fixing, or if you'd rather just close the pull request without merging my changes. No hard feelings if you do.

allnash commented:
I have merged this to my fork. Thank. Sorry for chiming in 2018. 😆

ghost commented:
Hah! no worries, don't even work at that job any more.

On Tue, Feb 6, 2018, 12:49 PM Nash Gadre <notifications@github.com> wrote:

> I have merged this to my fork. Thank. Sorry for chiming in 2018. 😆
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/dciarletta/d3-floorplan/pull/1#issuecomment-363506684>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAKQDaf9reeQ8yw14U04BMyPKC5Oj8yzks5tSJCkgaJpZM4EGPR4>
> .
>

allnash commented:
@AndrewArace check this out. I redid parts of it remove layering. Created some dynamic components.

https://github.com/allnash/d3-v4-floorplan

allnash commented:
the stupid part was d3 v4 API differences.
